lib/transfer.rb

```diff
@@ -0,0 +1,72 @@
+class Transfer
+  include ActiveModel::Validations
+
+  attr_accessor :amount, :category_id, :comment,
+    :bank_account_id, :reference_id, :comission
+
+  validates :amount, presence: :true, numericality: { greater_than: 0 }
+  validates_presence_of :bank_account_id
+  validates :reference_id, presence: true
+  validates :comission, numericality: { greater_than: 0 }
+  validate :transfer_amount
+  validate :transfer_currency
+  validate :transfer_account
+
+  def initialize(attributes = {})
+    attributes.each do |name, value|
+      send("#{name}=", value)
+    end
+  end
+
+  def save(transaction)
+    if valid?
+      out_transaction = Transaction.create(
+        amount: amount.to_f + comission.to_f, bank_account_id: bank_account_id,
```

Comment:
@Gataniel make `amount` and `commission` as money fields. Don't use `to_f` for money. It's possible to lose data with it.